backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/RoutingKey.java

```diff
@@ -0,0 +1,66 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.es;
+
+import java.util.Objects;
+
+import org.elasticsearch.common.Strings;
+
+import com.google.common.base.Preconditions;
+
+public class RoutingKey {
+    public interface Factory<T> {
+        RoutingKey from(T t);
+    }
+
+    public static RoutingKey fromString(String value) {
+        return new RoutingKey(value);
+    }
+
+    public static RoutingKey useDocumentId(DocumentId documentId) {
```

Comment:
untested method?

Reply:
btw it is only used in test code, is it really needed (cannot you just define a Factory<DocumentId> in the test code?)?